### ISSUE TITLE
Sand Island spawning at each stronghold

### ISSUE THREAD

**Issue by mirko93s** [BluePsychoRanger/SkyBlock_Collection#20]
![2024-06-15_14 26 56](https://github.com/BluePsychoRanger/SkyBlock_Collection/assets/21228368/cd801b48-d9f2-4120-8053-7da0a227214f)

Using 1.20.6 Fabolosuly Optimized + Skyblock - Standard, Advancements and Sand Island, all of them in the packaged (jar) version, no other mods and it is a new clean mc instance.

1 sand island with a barrier block will spawn above each stronghold as you can see in the screenshot above.

Also the initial sand island spawns right below the initial dirt/grass island, I don't think this is intended as well, this second issue doesn't always happen, and I wasn't able to find a reliable way to replicate it so I don't have a screenshot at the moment.

Tried using datapacks instead of jars, in the first world all was fine and no additional sand islands were generated above strongholds, but in the second world I created the sand island failed to generate at all, even the initial and correct one.

I don't know what is the issue, but it is surely related to the sand island, if I can help with more info/debug let me know.

**Comment by BluePsychoRanger** [comment]
I'll look into it. Do you still have the seed of that world with the barrier block?

**Comment by mirko93s** [comment]
Hi sorry for the late reply but unfortunately I don't have a certain seed because I experienced that pretty much consistently with every world I created while trying to get the mods to work.
At the end I only used the skyblock datapack (not the mod) and without the advancement/sand island addons. This worked 100% of the times. My guess is that the issue was in the sand island addon as it was the only thing that kept spawning at the wrong location or failing to spawn at all sometimes.
I hope this helps to narrow down the problem. I'll try to get the bug again, if possible and as soon as I have some spare time.

**Comment by BluePsychoRanger** [comment]
I can't reproduce the issue of the sand island spawning at each stronghold with the latest (beta) version of 1.20.6 fabulously optimized (5.12.0-beta.10 for 1.20.6). I've tried with 5 different worlds and none of them had a sand island above the stronghold. My guess is that one of the mods from Fabulously Optimized messed with structure generation, causing that bug. Do you know specifically what version of FO you were using? Maybe it was fixed in a more recent beta version for 1.20.6.

The initial sand island spawning directly under the spawn island is unfortunately a known issue and something that I've been unable to fix due to the way it needs to be generated.

**Comment by mirko93s** [comment]
I just tried myself hoping to get the same result but I didn't. I tried both the mods/packaged versions and the datapacks with multiple worlds but they both worked fine. Maybe it really was one of the mods from FO as you suggested.

I don't know the exact version because I don't have the instance in my launcher anymore but it is older than beta.6 for sure as it is the one I'm currently using to play with your datapack and the one I just tried to replicate the bug with but I failed to do so.

I'm sorry I can't help you further, if there are more questions or anything I can do to help, let me know, but I honestly think it was just the modpack at this point. I have been using your datapack for a while now and I didn't get any issue at all 😄
